### PR TITLE
BroadcastReceiver_Complete

### PR DESCRIPTION
**Pull request by AVI5HEK**
# Description

## Completed
1. Set alarm time
2. Choose multi days of week to alarm
3. Choose ring ton alarm
4. Cancel alarm
5. On/Off alarm
6. Edit/delete alarm
7. View list alarm

## Things to do next
1. Set alarm time to Google calendar
2. User can control alarm (remove or edit his alarm (schedule, color...)  from Google calendar)

### CODE REVIEW

File: app/src/main/java/com/framgia/alarm/RingAlarmActivity.java [hunk 1]
```diff
@@ -27,12 +28,11 @@ protected void onCreate(Bundle savedInstanceState) {
         setContentView(R.layout.activity_ring_alarm);
         Toolbar toolbar = (Toolbar) findViewById(R.id.toolbar);
         setSupportActionBar(toolbar);
-        Toast.makeText(getApplicationContext(), "working", Toast.LENGTH_LONG).show();
         mTime = (TextView) findViewById(R.id.text_time);
```
Comment (nvmanh) [inline app/src/main/java/com/framgia/alarm/RingAlarmActivity.java]: please break findViewById .... to another method, your oncreate method too long

File: app/src/main/java/com/framgia/alarm/RingAlarmActivity.java [hunk 2]
```diff
@@ -40,7 +40,20 @@ protected void onCreate(Bundle savedInstanceState) {
         mButtonCancel.setOnClickListener(new View.OnClickListener() {
             @Override
             public void onClick(View v) {
+                Toast.makeText(getApplicationContext(), getString(R.string
```
Comment (nvmanh) [inline app/src/main/java/com/framgia/alarm/RingAlarmActivity.java]: please implement your activity with OnClickListener interface

**Comment by nvmanh** [comment]
at home screen when set alarm time, label please make all (SUN, MON....) single line. I check on my device with MON it's broke to 2 lines

**Comment by AVI5HEK** [comment]
# Description of BroadcastReceiver_Complete_Comments_resolved

## Finished
   
* start app at first time, please add guide how to use this app, not show button (+) only, user only see blank screen and do not know what this app is
* on set alarm screen:
 * make ON default, not OFF
* Home screen with list alarms:
 * user can click on/off button to change status directly

## ToDo
* at home screen when set alarm time, label please make all (SUN, MON....) single line. 
* add button to add this event (alarm) to google calendar
* click to item to edit as current

File: app/src/main/java/com/framgia/alarm/model/Alarm.java [hunk 3]
```diff
@@ -28,27 +28,27 @@ public Alarm(long time, int status, String label, String alarm_tone_uri, String
...
         return mStatus;
     }
 
-    public String getmLabel() {
+    public String getLabel() {
         return mLabel;
     }
 
-    public String getmAlarm_tone_uri() {
+    public String getAlarm_tone_uri() {
```
Comment (nvmanh) [inline app/src/main/java/com/framgia/alarm/model/Alarm.java]: name convention

File: app/src/main/java/com/framgia/alarm/model/Alarm.java [hunk 4]
```diff
@@ -28,27 +28,27 @@ public Alarm(long time, int status, String label, String alarm_tone_uri, String
...
         return mAlarm_tone_uri;
     }
 
-    public String getmDay_schedule() {
+    public String getDay_schedule() {
```
Comment (nvmanh) [inline app/src/main/java/com/framgia/alarm/model/Alarm.java]: name convention

**Comment by nvmanh** [comment]
at home screen when set alarm time, label please make all (SUN, MON....) single line. I check on my device with MON it's broke to 2 lines ===> still error

**Comment by nvmanh** [comment]
Add to google calendar: 
- visible  button thí as DONE button not (more menu)
- notify user when add alarm to calendar success or not

File: app/src/main/res/menu/menu_set_alarm.xml [hunk 5]
```diff
@@ -1,10 +1,13 @@
...
-      xmlns:tools="http://schemas.android.com/tools">
+      xmlns:app="http://schemas.android.com/apk/res-auto">
     <item
         android:id="@+id/action_done"
         android:orderInCategory="100"
         android:title="@string/action_done"
         app:showAsAction="always"/>
+    <item
+        android:id="@+id/action_add_to_google_calendar"
+        android:title="@string/menu_set_alarm_add_to_calendar"
```
Comment (nvmanh) [inline app/src/main/res/menu/menu_set_alarm.xml]: text too long

**Comment by nvmanh** [comment]
i set alarm and it doesn't vibrate when alarm, and not play sound default when you do not choose rington

**Comment by nvmanh** [comment]
Set Alarm: create new alarm -> hide delete button please

**Pull request merged**